extension/otto/otto.go

```diff
@@ -171,15 +172,24 @@ func convertNilsToNulls(object interface{}) {
 //HandleEvent handles event
 func (env *Environment) HandleEvent(event string, context map[string]interface{}) (err error) {
 	vm := env.VM
+	var closeNotifier http.CloseNotifier
+	var closeNotify <-chan bool
+	if httpResponse, ok := context["http_response"]; ok {
+		if closeNotifier, ok = httpResponse.(http.CloseNotifier); ok {
+			closeNotify = closeNotifier.CloseNotify()
+		}
+	}
 	context["event_type"] = event
-	var halt = fmt.Errorf("exceed timeout for extension execution")
+	var timeout = fmt.Errorf("exceed timeout for extension execution")
+	var disconnected = fmt.Errorf("Client disconnected")
 
 	defer func() {
 		if caught := recover(); caught != nil {
 			if caughtError, ok := caught.(error); ok {
-				if caughtError.Error() == halt.Error() {
-					log.Error(halt.Error())
-					err = halt
+				if caughtError.Error() == timeout.Error() ||
```

Comment:
I agree, refactor should be in place, but this fix requires minimal changes to enable proper Interrupts. In case of select, I'd be required to repeat myself.